File: frontend/src/types/workspace.ts
export type TagType = 'success' | 'warning' | 'info' | 'primary' | 'danger'

export interface WorkspaceImportStep {
  index: string
  title: string
  description: string
  status: string
  tag_type: TagType
  action?: string
}

export interface WorkspaceConflictItem {
  id: string
  category: ConflictCategory
  severity: ConflictSeverity
  confidence: number
  level: string
  type: TagType
  title: string
  description: string
  affected_files: string[]
  status: string
}

export type ConflictSeverity = 'high' | 'medium' | 'low'
export type ConflictCategory =
  | 'paper_consistency'
  | 'behavior_regression'
  | 'trace_invalidation'
  | 'trace_coverage'
  | 'configuration_risk'

export type ConflictAnalysisStepStatus = 'running' | 'completed' | 'failed'

export interface ConflictAnalysisStep {
  step: number
  tool_name: string
  title: string
  summary: string
  status: ConflictAnalysisStepStatus
}

export interface WorkspaceChangeSummary {
  repository_revision: number
  analysis_status: string
  analysis_current: boolean
  has_changes: boolean
  changed_file_count: number
  changed_line_count: number
  latest_conflict_job_id: string | null
  latest_conflict_artifact_id: string | null
  analyzed_revision: number | null
  report_stale: boolean
}

export interface ConflictCodeEvidence {
  side: 'before' | 'after'
  path: string
  line_start: number
  line_end: number
  quote: string
}

export interface ConflictPaperEvidence {
  block_id: string
  quote: string
  span_id?: string
  char_start?: number
  char_end?: number
  quote_hash?: string
  page?: number | null
  association: 'trace' | 'inferred'
}

export interface ConflictTraceReference {
  trace_id: string
}

export interface ConflictReportItem {
  id: string
  category: ConflictCategory
  severity: ConflictSeverity
  confidence: number
  title: string
  description: string
  affected_files: string[]
  change_evidence: ConflictCodeEvidence[]
  affected_symbols: string[]
  callers: string[]
  graph_node_ids: string[]
  trace_refs: ConflictTraceReference[]
  paper_evidence: ConflictPaperEvidence[]
  recommendations: string[]
  verification_steps: string[]
}

export interface ConflictReport {
  schema_version: 'conflict-agent-v1'
  language?: 'zh-CN'
  repository_revision: number
  overall_risk: ConflictSeverity
  summary: {
    high: number
    medium: number
    low: number
    total: number
    changed_files: number
    changed_lines: number
  }
  items: ConflictReportItem[]
  unresolved: string[]
}

export interface WorkspaceReportCard {
  value: string
  title: string
  description: string
}

File: FinalRelease/source_code/backend/app/schemas/workspace.py
from pydantic import BaseModel, Field

from app.schemas.papers import WorkspacePaperPage
from app.schemas.repositories import (
    WorkspaceCodeFileRead,
    WorkspaceCodeTreeNode,
    WorkspaceTensorFlowRead,
)
from app.schemas.traces import WorkspaceTraceRow


class WorkspaceImportStep(BaseModel):
    index: str
    title: str
    description: str
    status: str
    tag_type: str
    action: str | None = None


class WorkspaceFlowNode(BaseModel):
    stage: str
    title: str
    description: str


class WorkspaceConflictItem(BaseModel):
    id: str = ""
    category: str = "behavior_regression"
    severity: str = "low"
    confidence: float = 0
    level: str
    type: str
    title: str
    description: str
    affected_files: list[str]
    status: str


class WorkspaceChangeSummary(BaseModel):
    repository_revision: int
    analysis_status: str
    analysis_current: bool
    has_changes: bool
    changed_file_count: int
    changed_line_count: int
    latest_conflict_job_id: str | None = None
    latest_conflict_artifact_id: str | None = None
    analyzed_revision: int | None = None
    report_stale: bool = False


class WorkspaceReportCard(BaseModel):
    value: str
    title: str
    description: str


class WorkspaceRead(BaseModel):
    project_id: str
    project_name: str
    import_steps: list[WorkspaceImportStep]
    paper_pages: list[WorkspacePaperPage]
    code_tree: list[WorkspaceCodeTreeNode]
    code_files: list[WorkspaceCodeFileRead]
    trace_rows: list[WorkspaceTraceRow]
    flow_nodes: list[WorkspaceFlowNode]
    tensor_flow: WorkspaceTensorFlowRead
    conflict_items: list[WorkspaceConflictItem]
    report_cards: list[WorkspaceReportCard]


class WorkspaceAnalyzeRequest(BaseModel):
    mode: str = Field(default="full", max_length=64)
    targets: list[str] = Field(default_factory=list)


class WorkspaceAnalyzeJob(BaseModel):
    project_id: str
    job_id: str
    status: str
    message: str
